crates/aptos/src/move_tool/mod.rs

```diff
@@ -748,7 +752,68 @@ impl CliCommand<TransactionSummary> for PublishPackage {
                 MAX_PUBLISH_PACKAGE_SIZE, size
             )));
         }
-        profile_or_submit(payload, &txn_options).await
+        // If JSON output file specified, store entry function JSON file on disk.
+        if let Some(output_file) = json_output_file {
+            // Extract entry function data from publication payload.
+            let entry_function = payload.into_entry_function();
+            let entry_function_id = EntryFunctionId {
+                module: MoveModuleId::from(entry_function.module().clone()),
+                name: IdentifierWrapper::from(entry_function.function()),
+            };
+            let package_metadata_hex = HexEncodedBytes(metadata_serialized).to_string();
+            let package_code_hex_vec: Vec<String> = compiled_units
+                .clone()
+                .into_iter()
+                .map(|element| HexEncodedBytes(element).to_string())
+                .collect();
+            // Construct entry function JSON file representation from entry function data.
+            let json = EntryFunctionArgumentsJSON {
+                function_id: entry_function_id.to_string(),
+                type_args: vec![],
+                args: vec![
+                    ArgWithTypeJSON {
+                        arg_type: "hex".to_string(),
+                        arg_value: serde_json::Value::String(package_metadata_hex),
+                    },
+                    ArgWithTypeJSON {
+                        arg_type: "hex".to_string(),
+                        arg_value: json!(package_code_hex_vec),
+                    },
+                ],
+            };
+            // Create save file options for checking and saving file to disk.
+            let save_file = SaveFile {
+                output_file,
+                prompt_options: txn_options.prompt_options,
+            };
+            save_file.check_file()?;
+            save_file.save_to_file(
+                "Publication entry function JSON file",
+                serde_json::to_string_pretty(&json)
+                    .map_err(|err| CliError::UnexpectedError(format!("{}", err)))?
+                    .as_bytes(),
+            )?;
+            Ok(TransactionSummary {
```

Comment:
This tells me tha this should not be part of this subcommand and instead a separate subcommand like `build-publish-payload`, I feel like returning a TransactionSummary for something that never submitted a txn will be confusing.

Reply:
https://github.com/aptos-labs/aptos-core/blob/6aa74479f6b3f2b44f06cc0aea7efbd9acff1a51/crates/aptos/src/move_tool/mod.rs#L782-L787